lib/themes/dosomething/paraneue_dosomething/js/takeover/Takeover.js

```diff
@@ -1,80 +1,91 @@
+  constructor($takeoverContainer) {
+    // @TODO - grab from page.
+    this.template = takeoverTpl;
+    this.baseUrl  = window.location.origin;
+    this.spinner  = '<div class="spinner"></div>';
+    this.activeClass = 'is-active';
 
-  activeClass: 'is-active',
-  /**
-   * Initialize the Signup interface.
-   *
-   * @param  {jQuery} $takeoverContainer the element to add the taked over to.
-   */
-  init: function($takeoverContainer = $('.takeover-container')) {
     const data = {
-      'header': 'Thanks for signing up',
       'shareLink': Drupal.settings.dosomethingUser.shareLink,
       'shareButtonMarkup': Drupal.settings.dosomethingUser.shareButtonMarkup,
     }
 
-    // Add the takeover to the page.
-    const slide = $takeoverContainer.hasClass('-no-slide') ? false : true;
-    this.addTakeover($takeoverContainer, data, slide);
+    this.addTakeover($takeoverContainer, data);
 
-    // Prepare validation fields that got entered into the DOM after page load.
-    Validation.prepareFields($('input'));
-
-    // Set up interactions
+    // Initialize screens in the experience.
     this.screens = $('.takeover__screen');
+    this.screenInit(this.screens);
+  }
+
+  screenInit($screens) {
     this.screenIndexes = this.setScreenIndexes();
     this.firstScreen = this.screens.first();
     this.lastScreen = this.screens.last();
-    this.firstScreen.addClass(this.activeClass);
-
-    // Set up submit handlers.
-    this.submitButton = $('.submit-postal-code');
-    this.handleSubmit(this.submitButton, {
-      'button' : this.submitButton,
-      'method' : 'GET',
-      'url'    : 'http://jsonplaceholder.typicode.com/posts/1',
-      'data'   : null,
-    });
-  },
+    this.firstScreen.addClass('is-active');
+  }
 
   /**
    * Adds the takeover to the page with or without a slide animation.
    *
    * @param  {jQuery} $container  the container to add the takeover to.
    * @param  {object} $data       An object containing the data to inject into the templates
-   * @param  {bool}   slide       Whether or not to add the takeover with a slide animation.
    */
-  addTakeover: function($container, data, slide = false) {
-    const markup = template(takeoverTpl);
+  addTakeover($container, data) {
+    const markup = template(this.template);
 
     $container.html(markup(data));
+
     this.takeoverContent = $('.takeover');
+  }
+
+  /*
+   * Store the indexes of each of the screens provided in the template.
+   */
+  setScreenIndexes() {
+    let indexes = [];
+
+    this.screens.each((key, scr) => indexes.push($(scr).data('index')));
+
+    return indexes;
+  }
+
+  /**
+   * Handles makeing a new scren active based on the direction passed.
+   *
+   * @param  {string}   direction
+   */
+  changeScreens(direction) {
+    // Check if an allowed direction was passed in.
+    if ($.inArray(direction, ['next', 'previous']) === -1) {
+      return;
+    }
+
+    const $current = $(`.takeover__screen.${this.activeClass}`);
```

Comment:
It could be nice to store the container (like as `this.$el` or something like that) when instantiating this class and make DOM lookups relative to that. Probably a little performance boost, and reduces the risk of there somehow being a conflicting classname elsewhere.